### DISCUSSION TITLE
Saving Texture Data to Disk

### DISCUSSION THREAD

**Discussion by RobertMtx** [DiligentGraphics/DiligentEngine#258] [Q&A] [answered]
What would be the simplest method to use in order to save the back buffer data as a png or other format on disk?

Is there a sample or tutorial that does anything similar? Possibly something that extracts texture data in a way that could be used as a half-guide? In my older DX9 engine, I had to lock the surface buffer and access the pixel data, but things have changed quite a lot since then.

Really appreciate any advice on this task.

**Comment by RobertMtx** [comment]
After posting this, I continued searching and stumbled onto `context::MapTextureSubresource()`, which searching for lead me to `SampleApp::SaveScreenCapture()`. I doubt I could ask for a better guide than this. Let me know if I'm mistaken here. Thanks.

**Reply by TheMostDiligent** [reply, marked as answer]
Yes, this is exactly the right place to look.

**Reply by RobertMtx** [reply]
Thanks, I appreciate it. So far so good. I just wish it had a resizing option! I'm currently using the NVidia texture library to convert and resize images, but injecting it in between the processes is likely to really slow things down. Not really a big deal for screen captures, but doing any type of video capture will likely require a more sophisticated setup. To be honest, I haven't tested purely resizing (without compression) with the library, so it may not be that bad.
